src/blackout.c

```diff
@@ -0,0 +1,225 @@
+    InitBgFromTemplate(bgConfig, 3, &Blackout_BgTemplate, GF_BG_TYPE_TEXT);
+    GfGfxLoader_GXLoadPal(NARC_graphic_font, 7, GF_BG_LYR_MAIN_0, 0x1A0, 0x20, HEAP_ID_FIELD); //TODO: update these enums
+    BG_SetMaskColor(3, RGB_WHITE);
+}
+
+static void Blackout_DrawMessage(FieldSystem *fieldSystem, TaskManager *taskManager) {
+    BlackoutScreenEnvironment *env = AllocFromHeap(HEAP_ID_FIELD, sizeof(BlackoutScreenEnvironment));
+
+    GF_ASSERT(env != NULL);
+    memset(env, 0, sizeof(BlackoutScreenEnvironment));
+    env->state = 0;
+    env->fieldSystem = fieldSystem;
+    env->bgConfig = BgConfig_Alloc(HEAP_ID_FIELD);
+    sub_0200FBF4(PM_LCD_TOP, RGB_WHITE); //are RGBs correct here?
+    sub_0200FBF4(PM_LCD_BOTTOM, RGB_WHITE);
+    sub_0200FBDC(0); //PM_LCD_TOP?
+    sub_0200FBDC(1); //PM_LCD_TOP?
+    Blackout_InitDisplays(env->bgConfig);
+    env->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, NARC_msg_msg_0203_bin, HEAP_ID_FIELD);
+    env->msgFmt = MessageFormat_New(HEAP_ID_FIELD);
+
+    AddWindow(env->bgConfig, &env->window, &Blackout_WindowTemplate);
+
+    BufferPlayersName(env->msgFmt, 0, Save_PlayerData_GetProfileAddr(FieldSystem_GetSaveData(fieldSystem)));
+    if (fieldSystem->location->mapId == MAP_T20R0201) {
+        // {STRVAR_1 3, 0, 0} scurried back\nhome, protecting the exhausted\nand fainted Pokémon from further\nharm...
+        Blackout_PrintMessage(env, msg_0203_00004, 0, 0);
+    } else {
+        // {STRVAR_1 3, 0, 0} scurried to\na Pokémon Center, protecting\nthe exhausted and fainted\nPokémon from further harm...
+        Blackout_PrintMessage(env, msg_0203_00003, 0, 0);
+    }
+    CopyWindowToVram(&env->window);
+    TaskManager_Call(taskManager, Task_ShowPrintedBlackoutMessage, env);
+}
+
+static BOOL Task_ShowPrintedBlackoutMessage(TaskManager *taskManager) {
+    BlackoutScreenEnvironment *env = TaskManager_GetEnvironment(taskManager);
+    switch (env->state) {
+        case 0:
+            BeginNormalPaletteFade(3, 1, 43, RGB_WHITE, 8, 1, HEAP_ID_32);
+            G2_BlendNone();
+            env->state++;
+            break;
+        case 1:
+            if (IsPaletteFadeFinished()) {
+                env->state++;
+            }
+            break;
+        case 2:
+            if (gSystem.newKeys & PAD_BUTTON_A || gSystem.newKeys & PAD_BUTTON_B || gSystem.touchNew != 0) {
+                BeginNormalPaletteFade(0, 0, 0, RGB_BLACK, 8, 1, HEAP_ID_32);
+                env->state++;
+            }
+            break;
+        case 3:
+            if (IsPaletteFadeFinished()) {
+                FillWindowPixelBuffer(&env->window, 0);
+                env->state++;
+            }
+            break;
+        case 4:
+            ClearFrameAndWindow2(&env->window, FALSE);
+            RemoveWindow(&env->window);
+            MessageFormat_Delete(env->msgFmt);
+            DestroyMsgData(env->msgData);
+            FreeBgTilemapBuffer(env->bgConfig, GF_BG_LYR_MAIN_3);
```

Comment:
Might need to clean up the Bg Config names as a whole but this function seems like it could be better named to like `BgConfig_FreeTilemapBuffer` or something similar

Reply:
out of scope atm